File: backend/models/project.py
"""Project domain data stored by PostgreSQL."""

from dataclasses import dataclass
from datetime import datetime


@dataclass(frozen=True)
class Project:
    id: int
    repository_url: str
    requested_revision: str
    worker_count: int
    commit_sha: str | None
    token_present: bool
    created_at: datetime
    manager_wake_at: datetime | None
    manager_wake_reason: str | None
    error: str | None

File: frontend/src/models/project.ts
export interface Project {
  id: string;
  repository_url: string;
  requested_revision: string;
  worker_count: number;
  commit_sha: string | null;
  token_present: boolean;
  created_at: string;
  error: string | null;
}

export interface CreateProjectRequest {
  repository_url: string;
  revision: string;
  worker_count: number;
  repository_token?: string;
}

export const MAX_WORKER_COUNT = 2_147_483_647;
